src/App.js

```diff
@@ -1,19 +1,179 @@
 import React from 'react';
-import './App.css';
+import { TodoList } from './components/TodoList';
+import './App.scss';
 
+import todos from './api/todos';
 import users from './api/users';
 
-function App() {
+function findUser(usersList, id) {
   return (
-    <div className="App">
-      <h1>Add todo form</h1>
-
-      <p>
-        <span>Users: </span>
-        {users.length}
-      </p>
-    </div>
+    usersList.find(user => user.id === id)
   );
 }
 
+const preparedTodos = todos.map(todo => (
+  {
+    ...todo,
+    user: findUser(users, todo.userId),
+  }
+));
+
+class App extends React.PureComponent {
+  state = {
+    currentTodosList: preparedTodos,
+    maxTitleLength: 15,
+    title: '',
+    user: '',
+    titleError: false,
+    userError: false,
+  }
+
+  handleSubmit = (event) => {
+    event.preventDefault();
+
+    if (!this.state.title || !this.state.user) {
+      if (!this.state.title) {
+        this.setState({ titleError: true });
+      }
+
+      if (!this.state.user) {
+        this.setState({ userError: true });
+      }
+
+      return;
+    }
+
+    this.setState((state) => {
+      const newId = Math.max(
+        ...state.currentTodosList.map(todo => todo.id),
+      ) + 1;
+
+      const newTodo = {
+        userId: +state.user,
+        id: newId,
+        title: state.title,
+        completed: false,
+        user: findUser(users, +state.user),
+      };
+
+      return {
+        currentTodosList: [
+          ...state.currentTodosList,
+          newTodo,
+        ],
+        title: '',
+        user: '',
+        titleError: false,
+        userError: false,
+      };
+    });
+  }
+
+  handleChange = (event) => {
+    const { name, value } = event.target;
+    const { title, maxTitleLength } = this.state;
+
+    if (name === 'title'
+      && title.length === maxTitleLength
+      && event.nativeEvent.data !== null
```

Comment:
for what you need this check?

Reply:
When we achieve max length without this checking I can't even press the backspace. So I added this checking.